### PR TITLE
Fixed a bug, which prevented fieldtype options from being set to a string "value"

### PR DESCRIPTION
**Pull request by LostKobrakai**

### CODE REVIEW

File: wire/modules/Fieldtype/FieldtypeOptions/FieldtypeOptions.module [hunk 1]
```diff
@@ -171,8 +171,9 @@ class FieldtypeOptions extends FieldtypeMulti implements Module {
 			
 		} else if(is_string($value)) {
 			// may be option 'title' (first) or option 'value' (second)
+			$_value = $value;
 			$value = $this->manager->getOptions($field, array('title' => $value));
```
Comment (LostKobrakai) [inline wire/modules/Fieldtype/FieldtypeOptions/FieldtypeOptions.module]: This line returned a `SelectableOptionArray`, which was used by this line ([SelectableOptionManager.php#L132](https://github.com/ryancramerdesign/ProcessWire/blob/master/wire/modules/Fieldtype/FieldtypeOptions/SelectableOptionManager.php#L132)) as array key, as soon as the next line called `getOptions` again. Using objects as array keys doesn't work, so it failed.

**Comment by ryancramerdesign** [comment]
@LostKobrakai do you know if this is still an issue? I seem to recall we fixed this a month or two ago, though perhaps in a different way.

**Comment by LostKobrakai** [comment]
It's still the same thing. `$this->manager->getOptions()` does return an empty options array (saved in $value) and the next line this empty thing is used to match by value instead of title.

**Comment by ryancramerdesign** [comment]
How do I observe the issue? Are you seeing an error thrown somewhere? Even
if something seems like an obvious bug or the like, I like to be able to
see it in action myself to make sure there isn't any other code that might
be relying on the incorrect result.

**Comment by LostKobrakai** [comment]
I was seeing it when trying to set a new field value by the option's value instead of the title. I've currently no fast way to check this, but I can describe what happened when I created the PR.
E.g.
```
1=first|First Item
2=second|Second Item
```
```php
$page->options = "first"; // somewhere while saving new data
```
The string will get to the sanitizer function as $value and will be first checked for 'title' matches. But as there are no matching titles `getOptions()` will return an empty options array, that does overwrite the to-be-sanitized string I wanted to set. The next line then does check if it can find matches by the option's values, but it's getting passed an already empty options array, therefore it won't find a match as well.

**Comment by LostKobrakai** [comment]
I just tested in on a nearly fresh installation and it's still not working. 

I created an options field and added it to the home template.
```
1=test|Test
2=nomore|Nicht mehr
```
After this I added this line in admin.php.
```php
$pages->get("/")->setAndSave("option", 'nomore'); // Didn't work
```
And then changed to this: 
```php
$pages->get("/")->setAndSave("option", 2); // Did work
```

**Comment by ryancramerdesign** [comment]
Thanks @LostKobrakai I have added this. I must have been running low on caffeine when I asked you about this before because it seems so obvious now. :) Thanks for your patience.

**Comment by LostKobrakai** [comment]
At least it's fixed now. Just a heads up isn't the same issue the case in line 188 of the file? No sure about it as I've not had the need to set multiple values by now.

**Comment by ryancramerdesign** [comment]
Thanks @LostKobrakai that was the same issue there too, and should already be fixed.

**Comment by LostKobrakai** [comment]
Great, now I can close this.

**Pull request closed without merging**